### DISCUSSION TITLE
Video: Migrating a Docker Sandbox to the Dockge app in Electric Eel

### DISCUSSION THREAD

**Discussion by mrstux** [Jip-Hop/jailmaker#247] [General]
G’day Guys, 

I made a video demonstrating the new Docker functionality in Electric Eel, and how to migrate a Docker sandbox to it. 

It’s a bit like a sequel to my Sandboxes with Jailmaker video 

Enjoy

[![TrueNAS Scale - Migrating Sandboxes to Docker in Electric Eel - YouTube Video](https://img.youtube.com/vi/R0Vdj1culo0/0.jpg)<br>Watch on YouTube](https://www.youtube.com/watch?v=R0Vdj1culo0 "TrueNAS Scale - Migrating Sandboxes to Docker in Electric Eel - YouTube Video")

**Comment by Lockszmith-GH** [comment]
Excellent walkthrough - your examples are very good.
I'd love to see some details on what is the architecture behind the new Electric Eel apps implementation. Seems to (finally) be very straight forward.

I wonder whether using the community apps would have any benefits on using docker-compose other than the UI.

When TrueCharts were steering the development, the interconnectedness of apps in their repo was amazing. I wonder if iX has any plans for such interconnectedness in their echo system.

Thanks for sharing. If you'd like another voice/face for an online discussion, I'd love to participate.

**Reply by mrstux** [reply]
> When TrueCharts were steering the development, the interconnectedness of apps in their repo was amazing. I wonder if iX has any plans for such interconnectedness in their echo system

I believe they do, but not in Electric Eel.

The next big pieces needed are Custom Labels support and Custom Networking support in order to make the built-in apps more useful for more complicated cases, but without that, anything done in regular docker can be done on Eel.

I could go into the architecture more, but at the end of the day, its docker, and by default the apps use a templated compose system in a hidden "ix-apps" dataset, which gets mounted to `/mnt/.ix-apps`. If you have any complicated requirements, due to not being able to customize the apps, you would want to use your own compose files, either via raw cli, or with dockge/portainer as a manager.

**Comment by airbornetrooper82573** [comment]
Just wanted to chime in, I upgraded to RC1 last week and since upgraded to RC2. When I upgraded jailmaker was no longer available with the command: jlkr shell docker. 

I'm not sure where TrueNAS moved the docker containers. I was able to get things back up and running since I had host paths but still curious where jailmaker items got moved to.

**Reply by mrstux** [reply]
most likely something to do with the alias being lost in your .zshrc, or the like.
